src/core/graphQL-interface/GraphQL.ts

```diff
@@ -33,15 +33,19 @@ function createResolvers(core: PerspectivismCore, config: OuterConfig) {
             agentByDID: async (parent, args, context, info) => {
                 checkCapability(context.capabilities, Auth.AGENT_READ_CAPABILITY)
                 const { did } = args;
-                const agentLanguage = core.languageController.getAgentLanguage().expressionAdapter;
-                if (!agentLanguage) {
-                    throw Error("Agent language does not have an expression adapter")
-                }
-                const expr = await agentLanguage.get(did);
-                if (expr != null) {
-                    return expr.data;
+                if (did != core.agentService.did) {
+                    const agentLanguage = core.languageController.getAgentLanguage().expressionAdapter;
+                    if (!agentLanguage) {
+                        throw Error("Agent language does not have an expression adapter")
+                    }
+                    const expr = await agentLanguage.get(did);
+                    if (expr != null) {
+                        return expr.data;
+                    } else {
+                        return null
+                    }
                 } else {
-                    return null
+                    return core.agentService.agent
```

Comment:
Is this needed for this feature for some reason, or just optimization on the side? I was wondering about scenarios where `core.agentService.agent` could get outdated compared to the value in the language, but I think it should be ok... ?

Reply:
Just an optimization. I don't think `core.agentService.agent` in its current form can become out of sync with the language since the agent service gets updated whenever we mutate the users profile via `agentUpdatePublicPerspective` & `agentUpdateDirectMessageLanguage`. But definitely something to keep an eye on if we change these functions at all.